Code/Framework/AzToolsFramework/AzToolsFramework/FocusMode/FocusModeInterface.h

```diff
@@ -36,6 +38,9 @@ namespace AzToolsFramework
         //! @return The entity id of the root of the Editor focus, or an invalid entity id if no focus is set.
         virtual AZ::EntityId GetFocusRoot(AzFramework::EntityContextId entityContextId) = 0;
 
+        //! Returns a list of the ids of all the entities that are descendants of the focus root.
+        virtual EntityIdList GetFocusedEntities(AzFramework::EntityContextId entityContextId) = 0;
```

Comment:
comment: I think this is fine for now, but we might want to consider in future making it possible to return a `const&` to avoid the copy... we'd likely want to investigate use cases first, but having this system 'own' the data and hand out a 'view' (read-only) might be useful.

Reply:
I was thinking about it, but the original draft didn't cache the data so it wasn't easy to do. 
We could do it, unless it ends up causing some weird loops (like a customer loops through it and does operations that cause the vector to be changed, destroying the iterators).

I will add a task with this list of improvements so we can investigate them with ease, but submit this first version asap so @jonawals can integrate it into his work.